install-feishin-appimage

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+arch=$(uname -m)
+
+if [ "$arch" != "x86_64" ] && [ "$arch" != "aarch64" ]; then
+	echo "CPU architecture not recognised (not x86_64 or aarch64). Aborting."
+	exit 1
+fi
+
+# workaround if we're not renaming the artifact
+if [ "$arch" = "aarch64" ]; then
+	arch='arm64'
+fi
+
+localShare="$HOME/.local/share"
+localShareIcons="${localShare}/icons/hicolor"
+
+if [ "$1" = "remove" ]; then
+	rm -v \
+		"${localShareIcons}"/512x512/apps/org.jeffvli.feishin.png \
+		"${localShareIcons}"/256x256/apps/org.jeffvli.feishin.png \
+		"${localShareIcons}"/128x128/apps/org.jeffvli.feishin.png \
+		"${localShareIcons}"/64x64/apps/org.jeffvli.feishin.png \
+		"${localShareIcons}"/32x32/apps/org.jeffvli.feishin.png \
+		"${localShare}"/applications/org.jeffvli.feishin.desktop \
+		./Feishin-linux-${arch}.AppImage
+	exit 0
+fi
+
+curl -L --create-dirs --write-out '%{filename_effective}\n' \
+	-O "https://github.com/jeffvli/feishin/releases/latest/download/Feishin-linux-${arch}.AppImage" \
+	-o "${localShareIcons}"/512x512/apps/org.jeffvli.feishin.png 'https://github.com/jeffvli/feishin/blob/development/assets/icons/512x512.png?raw=true' \
+	-o "${localShareIcons}"/256x256/apps/org.jeffvli.feishin.png 'https://github.com/jeffvli/feishin/blob/development/assets/icons/256x256.png?raw=true' \
+	-o "${localShareIcons}"/128x128/apps/org.jeffvli.feishin.png 'https://github.com/jeffvli/feishin/blob/development/assets/icons/128x128.png?raw=true' \
+	-o "${localShareIcons}"/64x64/apps/org.jeffvli.feishin.png 'https://github.com/jeffvli/feishin/blob/development/assets/icons/64x64.png?raw=true' \
+	-o "${localShareIcons}"/32x32/apps/org.jeffvli.feishin.png 'https://github.com/jeffvli/feishin/blob/development/assets/icons/32x32.png?raw=true'
+chmod -v u+x ./Feishin-linux-${arch}.AppImage
+
+waylandFlags=''
+if [ "$1" = "wayland-native" ]; then
+	waylandFlags='--enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto'
+fi
+
+mkdir -pv "${localShare}"/applications
+cat > "${localShare}"/applications/org.jeffvli.feishin.desktop << EOL
+[Desktop Entry]
+Name=Feishin
+GenericName=Music player
+Exec=$(pwd)/Feishin-linux-${arch}.AppImage --no-sandbox $waylandFlags
```

Comment:
Why is `--no-sandbox` being passed in? This should not be required by default.

Reply:
Agreed, which is why I posed the question on Discord :P

Reply:
Also of note, `--no-sandbox` is different from the `sandbox` option. The [`--sandbox` option](https://www.electronjs.org/docs/latest/tutorial/sandbox#configuring-the-sandbox) disables the `renderer` sandbox, whereas `--no-sandbox` disables the sandbox for _all_ processes (including other processes).

`--no-sandbox` is necessary primarily for systems where unpriviliged usernamespaces (where `sysctl kernel.unprivileged_userns_clone` is 1), or Ubuntu with whatever apparmor nonsense it has.

I suppose, you could try to detect this variable and then if it's zero, then add no sandbox, per https://docs.appimage.org/user-guide/troubleshooting/electron-sandboxing.html

Reply:
I guess I can check both `kernel.unprivileged_userns_clone == 0` and `kernel.userns_restrict == 1`.

The question is whether we really want to handle every possible permutation of this option right now, or if we just let it come to us down the line when it's actually needed.

Reply:
I think a check that's good _enough_ is probably fine. The most foolproof way would of course to just be to try to run the executable, and if it doesn't crash that's probably good enough, but that's not _great_.